frontend/src/ColumnSelector.js

```diff
@@ -1,80 +1,79 @@
-import React, { Component } from 'react';
-import Popover from './Popover';
-
+import React, { Component } from 'react'
+import Popover from './Popover'
 
 /**
  * Used to select the columns that appear in a table.
  */
 class ColumnSelector extends Component {
-    constructor(props) {
-        super(props);
+  constructor(props) {
+    super(props)
 
-        this.handleToggle = this.handleToggle.bind(this);
-        this.setAllColumns = this.setAllColumns.bind(this);
-        this.changeColumns = this.changeColumns.bind(this);
+    this.handleToggle = this.handleToggle.bind(this)
+    this.setAllColumns = this.setAllColumns.bind(this)
+    this.changeColumns = this.changeColumns.bind(this)
 
-        var defaultColumns = localStorage.getItem("meta-" + this.props.name);
-        if (defaultColumns) {
-            defaultColumns = JSON.parse(defaultColumns);
-        }
-        else {
-            defaultColumns = {
-                ...(props.type === "instructor" && {
-                    latest_semester: true,
-                    num_semesters: true,
-                }),
-                rInstructorQuality: true,
-                rCourseQuality: true,
-                rDifficulty: true,
-                rAmountLearned: true
-            };
-        }
-        this.defaultColumns = defaultColumns;
-        this.changeColumns(this.props.columns.map((a) => ({...a, show: a.required || !!defaultColumns[a.id]})));
+    let defaultColumns = localStorage.getItem(`meta-${this.props.name}`)
+    if (defaultColumns) {
+      defaultColumns = JSON.parse(defaultColumns)
+    } else {
+      defaultColumns = {
+        ...(props.type === 'instructor' && {
+          latest_semester: true,
+          num_semesters: true,
+        }),
```

Comment:
if `props.type !== 'instructor'` doesn't this say `...false` which would be an error? Probably will be more readable to split this into a few more functions / variable names

Reply:
split this into a seperate variable called instructorProps